spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/SymmetricKeyEncryptor.java

```diff
@@ -0,0 +1,29 @@
+package org.springframework.cloud.config.server.encryption;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.crypto.encrypt.TextEncryptor;
+
+/**
+ * Provide a default {@link TextEncryptorLocator} when a symmetric key is configured
+ *
+ * @author Ollie Hughes
+ *
+*/
+@Configuration
+@ConditionalOnProperty(value = "encrypt.key", matchIfMissing = false)
+public class SymmetricKeyEncryptor {
+
+	@Autowired
+	private TextEncryptor encryptor;
```

Comment:
The class name `SymmetricKeyEncryptor` implies it is something to do with symmetric keys, but the `@Autowired` will succeed for any `TextEncryptor`. I'm kind of worried that this is duplicating a code path in spring-cloud-commons somewhere.